chall07/edramire.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func logrequest(start time.Time, req string, res string, status int) {
+	duration := time.Since(start)
+	fmt.Printf("%d ms - GET %s\n", duration.Nanoseconds()/1000000, req)
+	if len(res) > 0 {
+		fmt.Printf("\t %d: %s\n", status, res)
+	}
+}
+
+func getrequest(request string) (string, int) {
+	res, err := http.Get(request)
+	if err != nil {
+		log.Fatal(err)
+	}
+	stringBody, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
```

Comment:
You should use the `defer` keyword to close `res.Body` no matter what happens. Here if `ioutil.ReadAll` panics for some reason, you will never close `res.Body`
```suggestion
	defer res.Body.Close()
	stringBody, err := ioutil.ReadAll(res.Body)
```
`defer` will automatically run the call it implies at the end of the function, it doesn't matter how it ends, it will call it.